src/tools/plan-generator.ts

```diff
@@ -50,23 +47,41 @@ export class PlanGenerator {
   async createPlans(): Promise<string[]> {
     const plans: string[] = [];
     for (const recipe of this.resolvedRecipes) {
-      const particles: string[] = [];
-      for (const particle of recipe.particles) {
-        particles.push(await this.createParticle(particle));
-      }
-
-      const planArgs = [ktUtils.listOf(particles)];
-      if (recipe.annotations.length > 0) {
-        planArgs.push(PlanGenerator.createAnnotations(recipe.annotations));
-      }
-
-      const start = `object ${recipe.name}Plan : `;
-      const plan = `${start}${ktUtils.applyFun('Plan', planArgs, {startIndent: start.length})}`;
-      plans.push(plan);
+      const plan = `${ktUtils.applyFun(`val ${recipe.name}Plan = Plan`, [
+        ktUtils.listOf(await Promise.all(recipe.particles.map(p => this.createParticle(p)))),
+        ktUtils.listOf(recipe.handles.map(h => this.handleVariableName(h))),
+        PlanGenerator.createAnnotations(recipe.annotations)
+      ])}`;
+
+      plans.push([
+        ...await Promise.all(recipe.handles.map(h => this.createHandleVariable(h))),
+        plan
+      ].join('\n'));
     }
     return plans;
   }
 
+  /**
+   * @returns a name of the generated Kotlin variable with a Plan.handle corresponding to the given handle.
+   */
+  handleVariableName(handle: Handle): string {
+    return `${handle.recipe.name}_Handle${handle.recipe.handles.indexOf(handle)}`;
+  }
+
+  /**
+   * Generates a variable for a handle to be places inside a `Plan.Particle` object.
+   * We generate a variable instead of a inlining a handle, as we want to reference
+   * a handle both in a list of all plan handles, as well as in individual handle connections.
+   */
+  async createHandleVariable(handle: Handle): Promise<string> {
+    handle.type.maybeEnsureResolved();
+    return ktUtils.applyFun(`val ${this.handleVariableName(handle)} = Handle`, [
```

Comment:
same as above.

Reply:
If you want a simpler interface, I can see value in creating a new function on ktUtils. WDYT?

Reply:
Done!